### ISSUE TITLE
AL07: Framework limitation prevents rule execution on CREATE TABLE statements

### ISSUE THREAD

**Issue by Fank** [quarylabs/sqruff#1713]
## Problem

The AL07 (aliasing.forbid) rule cannot be executed on CREATE TABLE statements due to a framework limitation, even though the rule logic has been enhanced to support this syntax.

## Current Behavior

```sql
-- This syntax is not linted by AL07
CREATE TABLE #TempTable
AS
(
  Select ColumnD
  from SchemaA.TableB AliasC  -- Alias not detected/flagged
  where ColumnD >= @VariableE
)
```

## Root Cause

1. **Framework design** - linter doesn't call rules for CREATE TABLE statements
2. **Rule crawling behavior** - CREATE TABLE statements bypass normal rule execution
3. **Segment analysis limitation** - some statement types excluded from rule processing

### Enhanced Rule Logic (Already Implemented)
- ✅ AL07 rule crawl_behaviour includes CreateTableStatement
- ✅ AL07 eval() method handles CREATE TABLE ... AS SELECT syntax  
- ✅ Rule correctly finds nested SELECT statements in CREATE TABLE
- ✅ Comprehensive test cases added for various CREATE TABLE scenarios

### Failing Tests (All marked as framework limitation)
- `issue_1639` - Basic CREATE TABLE with alias
- `test_tsql_create_table_with_multiple_aliases` - Multiple table aliases
- `test_tsql_create_table_temp_with_hash` - Temp table with # prefix
- `test_tsql_create_table_global_temp` - Global temp table with ## prefix  
- `test_tsql_create_table_with_cte` - CREATE TABLE with CTE

### Current Status
All tests marked as: `ignored: "AL07 rule not called for CREATE TABLE statements - framework limitation"`

## Framework Investigation Needed

Need to investigate why CREATE TABLE statements don't trigger rule execution:

### 1. Crawling Behavior
```rust
// AL07 correctly specifies it wants CREATE TABLE statements
fn crawl_behaviour(&self) -> Crawler {
    SegmentSeekerCrawler::new(
        const { SyntaxSet::new(&[SyntaxKind::SelectStatement, SyntaxKind::CreateTableStatement]) }
    )
    .into()
}
```

### 2. Rule Execution Framework
- Check if linter framework filters out CREATE TABLE before rule execution
- Verify SegmentSeekerCrawler handles CreateTableStatement correctly
- Investigate if there are statement type exclusions in rule processing

### 3. Parser Integration
- Confirm CREATE TABLE statements are properly parsed and segmented
- Verify segment type hierarchy for CREATE TABLE ... AS SELECT

### Phase 1: Framework Analysis
1. **Trace rule execution** for CREATE TABLE statements
2. **Debug SegmentSeekerCrawler** behavior with CreateTableStatement
3. **Check linter main loop** for statement type filtering
4. **Verify segment type registration** for CREATE TABLE

### Phase 2: Framework Enhancement
1. **Update framework** to include CREATE TABLE in rule execution
2. **Fix crawling behavior** if SegmentSeekerCrawler excludes CREATE TABLE
3. **Add framework tests** for CREATE TABLE rule execution
4. **Verify backward compatibility** with existing rules

### Phase 3: Validation
1. **Enable ignored tests** once framework is fixed
2. **Verify AL07 works correctly** on CREATE TABLE statements
3. **Test other rules** that might benefit from CREATE TABLE analysis
4. **Update documentation** for CREATE TABLE rule support

## Impact

- **Priority**: Medium - affects T-SQL CREATE TABLE alias enforcement
- **Scope**: Framework-level change required
- **Benefit**: Enables comprehensive alias checking in CREATE TABLE statements
- **Risk**: Low - rule logic already implemented and tested

## Success Criteria

When fixed, these patterns should be properly linted:
```sql
-- Should flag alias
CREATE TABLE result AS
SELECT col1 FROM table1 AS t1  -- AL07 should flag 't1' alias

-- Should flag multiple aliases  
CREATE TABLE result AS
SELECT a.col1, b.col2
FROM table1 AS a
JOIN table2 AS b ON a.id = b.id  -- AL07 should flag both aliases
```

## Labels
- `framework-limitation`
- `rule:AL07`
- `enhancement`
- `dialect:tsql`
- `create-table`
